examples/analyze_excel_ui.py

```diff
@@ -85,50 +85,153 @@
     # Part 2: Examples on how to use processed cell sample data
     ###
 
-    # We will read IPTG concentrations for each sample from the Excel file.
-    # ``samples_table`` contains all the data from sheet "Samples", including
-    # data not directly used by ``FlowCal.excel_ui.process_samples_table()``.
-    # ``samples_table`` is a pandas dataframe, with each column having the
-    # same name as the corresponding header in the Excel file.
-    # We multiply the IPTG concentration by 1000 to convert to micromolar.
-    iptg = samples_table['IPTG (mM)']*1000
-
-    # Histogram of all samples
-    # Here, we plot the fluorescence histograms of all five samples in the same
-    # figure, using ``FlowCal.plot.hist1d``. Note how this function can be used
+    # Each entry in the Excel table has a corresponding ID, which can be used
+    # to reference information associated with or the sample loaded from a
+    # row in the Excel file. Collect the IDs of the non-control samples (i.e.,
+    # 'S0001', ..., 'S0010').
+    sample_ids = ['S00{:02}'.format(n) for n in range(1,10+1)]
+
+    # We will read DAPG concentrations from the Excel file. ``samples_table``
+    # contains all the data from sheet "Samples", including data not directly
+    # used by ``FlowCal.excel_ui.process_samples_table()``. ``samples_table``
+    # is a pandas dataframe, with each column having the same name as the
+    # corresponding header in the Excel file.
+    dapg = samples_table.loc[sample_ids,'DAPG (uM)']
+
+    # Plot 1: Histogram of all samples
+    #
+    # Here, we plot the fluorescence histograms of all ten samples in the same
+    # figure using ``FlowCal.plot.hist1d``. Note how this function can be used
     # in the context of accessory matplotlib functions to modify the axes
-    # limits and labels and add a legend, among others.
+    # limits and labels and to add a legend, among other things.
+
+    # Color each histogram according to its corresponding DAPG concentration.
+    # Use a perceptually uniform colormap (cividis), and transition among
+    # colors using a logarithmic normalization, which comports with the
+    # logarithmically spaced DAPG concentrations.
+    cmap = mpl.cm.get_cmap('cividis')
+    norm = mpl.colors.LogNorm(vmin=1e0, vmax=500)
+    colors = [cmap(norm(dapg_i)) if dapg_i > 0 else cmap(0.0)
+              for dapg_i in dapg]
+
     plt.figure(figsize=(6,3.5))
-    FlowCal.plot.hist1d(list(samples.values()),
+    FlowCal.plot.hist1d([samples[s_id] for s_id in sample_ids],
                         channel='FL1',
                         histtype='step',
-                        bins=128)
-    plt.ylim([0, 2000])
+                        bins=128,
+                        edgecolor=colors)
+    plt.ylim((0,2500))
+    plt.xlim((0,5e4))
     plt.xlabel('FL1  (Molecules of Equivalent Fluorescein, MEFL)')
-    plt.legend(['{} $\mu M$ IPTG'.format(i) for i in iptg],
+    plt.legend(['{} $\mu M$ DAPG'.format(i) for i in dapg],
```

Comment:
Too many decimal places. Replace `{}` with `{:.1f}` or `{:.2f}`.

Reply:
Done in ab40d38358da8af0b33bba3c532ca8d5b4b173df.